File: Assets/Resources/Scripts/Grenade.cs
using UnityEngine;
using System.Collections;

public class Grenade : MonoBehaviour {

	public float radius = 2.0f;
	public float explosiveDelay = 5.0f;
	// Use this for initialization
	void Start () {
	
		 Vector3 grenadeOrigin = transform.position;

	}
	
	// Update is called once per frame
	void Update () {
	
	}

	IEnumerator ExplosionTimer()
	{
		yield return new WaitForSeconds(explosiveDelay);
	}
}

File: Assets/Resources/Scripts/Grenade.js
#pragma strict

var radius : float = 5.0;
var power : float = 10.0;

var explosiveDelay : float = 5.0;

function Start () {

yield WaitForSeconds(explosiveDelay);
var grenadeOrigin : Vector3 = transform.position;
var colliders : Collider[] = Physics.OverlapSphere (grenadeOrigin, radius);

for (var hit : Collider in colliders){

	if (hit.rigidbody)
	{
	
		hit.rigidbody.AddExplosionForce(power, grenadeOrigin, radius);
		Destroy(gameObject);
	
	}

}

}

function Update () {

}